### PR TITLE
10 Add RDM->CLM CLI migration path

### PR DESCRIPTION
**Pull request by johnatawnclementawn**
Fixes pg function to point to correct controlled list tables & columns.
Adds CLI tool for utilizing that pg function to migration collections to lists. 

Name of `commands/controlled_lists.py` is still up for debate.

### CODE REVIEW

File: tests/cli_tests.py [hunk 1]
```diff
@@ -3,10 +3,13 @@
 
 from django.conf import settings
```
Comment (jacobtylerwalls) [inline tests/cli_tests.py:4]: unused

File: tests/cli_tests.py [hunk 2]
```diff
@@ -3,10 +3,13 @@
...
 from django.core import management
+from django.urls import reverse
 from django.test import TestCase
+from django.test.client import Client
 from django.test.utils import captured_stdout
 
-from arches_references.models import List
+from arches_references.models import List, ListItem, ListItemValue
```
Comment (jacobtylerwalls) [inline tests/cli_tests.py:11]: ListItemValue unused

File: tests/cli_tests.py [hunk 3]
```diff
@@ -57,3 +60,41 @@ def test_import_controlled_list(self):
         self.assertTrue(List.objects.filter(pk=list_pk).exists())
 
     ### TODO Add test for creating new language if language code not in db but found in import file
+
```
Comment (jacobtylerwalls) [inline tests/cli_tests.py:63]: I missed this before, but in input_file above, avoid providing a slashed input as one of the args to os.path.join, which could fail on windows. Provide each path component as separate args to join().

File: tests/cli_tests.py [hunk 4]
```diff
@@ -57,3 +60,41 @@ def test_import_controlled_list(self):
...
+
+class RDMToControlledListsETLTests(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+
+        skos = SKOSReader()
+        rdf = skos.read_file(
+            os.path.join(PROJECT_TEST_ROOT, "data/concept_label_test_collection.xml")
```
Comment (jacobtylerwalls) [inline tests/cli_tests.py:72]: same

File: tests/cli_tests.py [hunk 5]
```diff
@@ -57,3 +60,41 @@ def test_import_controlled_list(self):
...
+            host="http://localhost:8000/plugins/controlled-list-manager/item/",
+            preferred_sort_language="en",
+            overwrite=False,
+            stdout=output,
+        )
+
+        self.assertTrue(List.objects.filter(name="Concept Label Import Test").exists())
+        self.assertTrue(
+            ListItem.objects.filter(id="89ff530a-f350-44f0-ac88-bdd8904eb57e").exists()
+        )
```
Comment (jacobtylerwalls) [inline tests/cli_tests.py:100]: Asserting against a UUID is not as informative as something like this:

```suggestion
        imported_list = List.objects.get(name="Concept Label Import Test")
        imported_items = imported_list.list_items.all()
        self.assertEqual(len(imported_items), 3)
```

File: tests/cli_tests.py [hunk 6]
```diff
@@ -57,3 +60,41 @@ def test_import_controlled_list(self):
...
+        client = Client()
+        client.login(username="admin", password="admin")
+        response = client.get(
+            reverse(
+                "make_collection",
+                kwargs={"conceptid": "7c90899a-dbe9-4574-9175-e69481a80b3c"},
+            )
+        )
+
+    def test_migrate_collections_to_controlled_lists(self):
```
Comment (jacobtylerwalls) [inline tests/cli_tests.py:85]: Another test that the failure path is gracefully handled would be nice.

File: arches_references/management/commands/controlled_lists.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        )
+
+        failed_collections = [
+            collection
+            for collection in collections_to_migrate
+            if collection not in collections_in_db
+        ]
+
+        if len(failed_collections) > 0:
+            self.stdout.write(
```
Comment (jacobtylerwalls) [inline arches_references/management/commands/controlled_lists.py:102]: use stderr?

File: arches_references/management/commands/controlled_lists.py [hunk 8]
```diff
@@ -0,0 +1,120 @@
+from arches.app.models.models import Concept, Value
```
Comment (jacobtylerwalls) [inline arches_references/management/commands/controlled_lists.py:1]: concept unused

File: arches_references/management/commands/controlled_lists.py [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+                % ", ".join(failed_collections)
+            )
+
+        if len(collections_in_db) > 0:
+            from django.db import connection
+
+            cursor = connection.cursor()
+            cursor.execute(
+                """
+                select * from __arches_migrate_collections_to_clm(
```
Comment (jacobtylerwalls) [inline arches_references/management/commands/controlled_lists.py:113]: Cool. In the future we can consider implementing the `Func` [expressions API](https://docs.djangoproject.com/en/5.0/ref/models/expressions/#func-expressions) to be able to use custom db functions with the ORM.

**Review by jacobtylerwalls — APPROVED** [review]
Looks good! Some minor feedback at your option.

**Review by jacobtylerwalls — APPROVED** [review]

**Comment by johnatawnclementawn** [comment]
Thanks for the review @jacobtylerwalls 👍🏼

**Pull request merged**